config/validate.go

```diff
@@ -218,9 +218,21 @@ func validateJobs(c atc.Config) ([]Warning, error) {
 			)
 		}
 
-		planWarnings, planErrMessages := validatePlan(c, identifier+".plan", atc.PlanConfig{Do: &job.Plan})
+		planInputs, planWarnings, planErrMessages := validatePlan(c, identifier+".plan", atc.PlanConfig{Do: &job.Plan})
```

Comment:
Could this be done by looping over `JobInputs(job)` instead of collecting them during recursion?